src/ws.cpp

```diff
@@ -91,3 +93,43 @@ void Ws::index(Context *c, const QString &hostId, const QString &uuid)
         sock->flush();
     });
 }
+
+void Ws::createSshTunnel(QAbstractSocket *sock, const QUrl &url, quint16 port) {
+    qCDebug(V_WS) << "Create SSH tunnel to"<< url.host() << "as" << url.userName();
+
+    QStringList args;
+    args << u"-TCv"_qs;
+    args << u"-L"_qs << QStringLiteral("127.0.0.1:%1:127.0.0.1:%1").arg(port);
+    args << u"-l"_qs << url.userName();
+    args << url.host();
+
+    auto process = new QProcess(sock);
+    connect(sock, &QAbstractSocket::disconnected, process, [process]{
+        if (process->state() == QProcess::Running) {
+            process->terminate();
+            qCDebug(V_WS) << "Shutdown ssh tunnel:" << process->waitForFinished();
+        }
+    });
+
+    process->setProgram(u"ssh"_qs);
+    process->setArguments(args);
+    process->setReadChannel(QProcess::StandardError);
+    process->start();
+
+    if (process->waitForStarted()) {
```

Comment:
in fact it would be good to avoid theses blocking calls, some of libvirt calls (which now iirc have async) block event loop which sucks a lot.

Reply:
Yeah, that would be the next step. But that would need a refactoring of the current code base.